src/main/java/fi/hsl/transitdata/cancellation/util/CacheUtils.java

```diff
@@ -110,7 +107,7 @@ public static List<CancellationData> handleBulletinCancellations(
         }
         
         // check
-        if (bulletinsCache.getIfPresent(bulletinId).keySet().size() != modifiedCancellationDataList.size()) {
+        if (Objects.requireNonNull(bulletinsCache.getIfPresent(bulletinId)).keySet().size() != modifiedCancellationDataList.size()) {
```

Comment:
I don't like requireNonNull here since all it does is throw NullPointerException is Object supplied is null.

Clean solution to avoid NPE would be to use .get instead of .getIfPresent and implement simple MappingFunction that returns empty map for bulletinId if such did not exist. Not 100% sure on side effects this has though, so think through them before proceeding.